imports/plugins/included/product-variant/components/productGrid.js

```diff
@@ -1,11 +1,43 @@
 import React, { Component } from "react";
 import PropTypes from "prop-types";
+import InfiniteScroll from "react-infinite-scroll-component";
 import { Components } from "@reactioncommerce/reaction-components";
 
 class ProductGrid extends Component {
   static propTypes = {
     products: PropTypes.array
   }
+  constructor(props) {
+    super(props);
+    this.state = {
+      allProducts: this.props.products,
+      nextProducts: [],
+      lastIndex: 12,
+      hasMore: true
+    };
+    this.generateProducts = this.generateProducts.bind(this);
+  }
+
+
+  componentWillReceiveProps(nextProps) {
+    if (this.state.allProducts.length > 12) {
+      this.setState({ nextProducts: nextProps.products.slice(0, this.state.lastIndex) });
+    } else {
+      this.setState({ nextProducts: this.state.allProducts });
+    }
+  }
+
+  generateProducts() {
```

Comment:
Write this as 
```js
generateProducts = () => {...}
```
then you won't have to bind the method in the constructor

Reply:
Refactored